### PR TITLE
Actor IT - timers, reminders, turn-based concurrency

### PR DESCRIPTION
**Pull request by LMWF**
IT for actor timer, reminder, turn-based concurrency

# Description

From header of the test:

  Create an actor, register a timer and a reminder, then invoke additional actor method calls.
    Validate turn-based concurrency by getting logs from the actor through an actor method, and asserting that:
      - "Enter" and "Exit" calls alternate
      - An entered actor method is exited before a subsequent actor method call.
    Unregister the timer and reminder and verify they do not fire subsequently.  Also verify the timer and reminder
    did fire a reasonable (with delta) number of times when they were registered.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk-tests/src/test/java/io/dapr/it/actors/ActorTurnBasedConcurrencyIT.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+      client.invokeAPI(DaprHttp.HttpMethods.DELETE.name(), url, null, null).block();
+    } catch(Exception e) {
+      // informational only
+      System.out.println("Caught " + e.toString());
+    }
+
+  }
+
+  @BeforeClass
+  public static void init() throws Exception {
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/actors/ActorTurnBasedConcurrencyIT.java]: Delete.

File: sdk-tests/src/test/java/io/dapr/it/actors/ActorTurnBasedConcurrencyIT.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+    // get history again, we don't additional timer/reminder calls
+    logs = getAppMethodCallLogs(proxy);
+    validateEventNotObserved(logs, "stopTimer", "clock");
+    validateEventNotObserved(logs, "stopReminder", "receiveReminder");
+  }
+
+  ArrayList<MethodEntryTracker> getAppMethodCallLogs(ActorProxy proxy) {
+    ArrayList<String> logs = proxy.invokeActorMethod("getCallLog", ArrayList.class).block();
+    System.out.println("here");
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/actors/ActorTurnBasedConcurrencyIT.java]: remove

File: sdk-tests/src/test/java/io/dapr/it/actors/ActorTurnBasedConcurrencyIT.java [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        currentMethodName = s.getMethodName();
+      } else {
+        assertTrue(currentMethodName.equals(s.getMethodName()));
+      }
+    }
+
+    boolean flag = true;
+    for (MethodEntryTracker s : logs) {
+      if (s.getIsEnter() == flag) {
+        ; // correct, flip it
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/actors/ActorTurnBasedConcurrencyIT.java]: remove

File: sdk-tests/src/test/java/io/dapr/it/actors/app/MyActor.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+package io.dapr.it.actors.app;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public interface MyActor {
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/actors/app/MyActor.java]: Can you give it a name that is specific to your test? Also, we are having those in services/springboot.

File: sdk-tests/src/test/java/io/dapr/it/actors/app/MyActorImpl.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+import io.dapr.actors.runtime.Remindable;
+import io.dapr.it.actors.MethodEntryTracker;
+import reactor.core.publisher.Mono;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.*;
+
+@ActorType(name = "MyActorTest")
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/actors/app/MyActorImpl.java]: Same for name.

File: sdk-tests/src/test/java/io/dapr/it/actors/app/TestController.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+import io.dapr.actors.runtime.ActorRuntime;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Mono;
+
+/**
+ * SpringBoot Controller to handle callback APIs for Dapr.
+ */
+@RestController
+class TestController {
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/actors/app/TestController.java]: For now, reuse the existing controller and add your actor type to the list. I have a small refactoring that will allow us to reuse DaprApplication.

File: sdk-tests/src/test/java/io/dapr/it/actors/app/TestApplication.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+package io.dapr.it.actors.app;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * Dapr's HTTP callback implementation via SpringBoot.
+ */
+@SpringBootApplication(scanBasePackages = {"io.dapr.it.actors.app"})
+public class TestApplication {
```
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/actors/app/TestApplication.java]: Let's reuse the existing DaprApplication.

File: sdk-tests/src/test/java/io/dapr/it/actors/app/TestController.java (hunk 6 above)
Comment (LMWF) [reply sdk-tests/src/test/java/io/dapr/it/actors/app/TestController.java]: The issue is the existing DaprApplication overrides the config call with non-default settings and I want default settings.

Comment (artursouza) [reply sdk-tests/src/test/java/io/dapr/it/actors/app/TestController.java]: OK.

**Pull request merged**